web_advanced_filters/model/ir_filters.py

```diff
@@ -65,17 +66,63 @@ def eval_n(domain):
                     cr, uid, this['union_filter_ids'], ['domain'],
                     context=context)])
             for c in self.read(cr, uid, this['complement_filter_ids'],
-                               ['domain'], context=context):
-                domain = expression.AND([
-                    domain,
-                    ['!'] + eval_n(c['domain'])])
-            result[this['id']] = str(domain)
+                               ['domain', 'evaluate_before_negate',
+                                'model_id'],
+                               context=context):
+                if c['evaluate_before_negate']:
+                    domain = expression.AND([
+                        domain,
+                        [
+                            [
+                                'id', 'not in',
+                                self.pool[c['model_id']].search(
+                                    cr, uid, eval_n(c['domain']),
+                                    context=context)
+                            ]
+                        ]])
+                else:
```

Comment:
ll.73..82 would be more readable if you stored the result of the call to search() in a variable and replace these 10 lines by 

```
domain = expression.AND([domain, [('id', 'not in', neg_ids)]])
```